ci/provisioning/post_provision_config_common_functions.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+
+set -eux
+
+: "${DAOS_STACK_RETRY_DELAY_SECONDS:=60}"
+: "${DAOS_STACK_RETRY_COUNT:=3}"
+: "${DAOS_STACK_MONITOR_SECONDS:=600}"
+: "${BUILD_URL:=Not_in_jenkins}"
+: "${STAGE_NAME:=Unknown_Stage}"
+: "${OPERATIONS_EMAIL:=$USER@localhost}"
+
+retry_dnf() {
+    local monitor_threshold="$1"
+    shift
+
+    local args=("dnf" "-y" "${@}")
+    local attempt=0
+    local rc=0
+    while [ $attempt -lt "${RETRY_COUNT:-$DAOS_STACK_RETRY_COUNT}" ]; do
+        if monitor_cmd "$monitor_threshold" "${args[@]}"; then
+            # Command succeeded, return with success
+            if [ $attempt -gt 0 ]; then
+                send_mail "Command retry successful in $STAGE_NAME after $((attempt + 1)) attempts using ${repo_servers[0]} as initial repo server " \
```

Comment:
(style)  line over 100 characters
(lint) repo_servers is referenced but not assigned (did you mean 'repo_server'?). [SC2154]